src/test/java/org/folio/rest/impl/CodexInstanceResourceImplTest.java

```diff
@@ -215,6 +214,66 @@ public void getCodexInstancesSuccessTest(TestContext context) {
     asyncLocal.complete();
   }
 
+  @Test
+  public void getCodexInstancesIdSearchSuccessTest(TestContext context) {
+    final Async asyncLocal = context.async();
+    logger.info("Testing for successful instance collection");
+
+    final Response r = RestAssured
+        .given()
+          .header(tenantHeader)
+          .header(urlHeader)
+          .header(contentTypeHeader)
+        .get(String.format("/codex-instances?query=id=%d", 99999))
+          .then()
+            .contentType(ContentType.JSON)
+            .log()
+            .ifValidationFails()
+            .statusCode(200).extract().response();
+
+    if(r != null) { //Ensure that the response is not null
+      final String body = r.getBody().asString();
+      final JsonObject json = new JsonObject(body);
+      //Ensure that total records and instances keys are present in response
+      context.assertEquals(json.getInteger("totalRecords"), Integer.valueOf(1));
+      context.assertTrue(json.containsKey("instances"));
+    }
+
+    // Test done
+    logger.info("Test done");
+    asyncLocal.complete();
+  }
+
+  @Test
+  public void getCodexInstancesIdSearchFailTest(TestContext context) {
+    final Async asyncLocal = context.async();
+    logger.info("Testing for successful instance collection");
+
+    final Response r = RestAssured
+        .given()
+          .header(tenantHeader)
+          .header(urlHeader)
+          .header(contentTypeHeader)
+        .get(String.format("/codex-instances?query=id=%d", 1))
+          .then()
+            .contentType(ContentType.JSON)
+            .log()
+            .ifValidationFails()
+            .statusCode(200).extract().response();
```

Comment:
@mreno-EBSCO : Is this supposed to still return a status code of 200 if the search fails?

Reply:
Yes. This is a search, so the return is an empty instance list. Same an searching by title with a title that matched nothing. It is not an error, but an empty list.

Reply:
Ah, gotcha 👍